### PR TITLE
Putio client refactor to address authorization issues

### LINKED ISSUE #3309: put.io authentication errors
### Before submitting your issue:

Enable debug logging in SickRage settings, reproduce the error (be sure to disable after the bug is fixed)

Branch/Commit:  66babc76d28646807c1040a3e225072b410489ad
OS: Ubuntu 
What you did: Search for episode
What happened: put.io authentication error
What you expected: snatch torrent and push to put.io
Logs:
```
2017-03-03 23:54:36 WARNING SEARCHQUEUE-MANUAL-153021 :: put_io: Authentication Failed
AA KeyError: 'location'
AA raise requests_exception
AA File "/opt/sickrage/sickbeard/helpers.py", line 1434, in handle_requests_exception
AA Traceback (most recent call last):
2017-03-03 23:54:36 ERROR SEARCHQUEUE-MANUAL-153021 :: [66babc7] Request failed: 'location'
```

I can't say when exactly this started happening, but my guess is ~2/28 (I update every other day or so)  
I tried revoking OAuth access and reauthorizing to fix, but the callback just goes to sickrage.github.io 

Client still works under "Manage Torrents" as I assume it doesn't use OAuth

### PR DESCRIPTION
**Pull request by ukanga**
Fixes #3309 and #2715

Proposed changes in this pull request:
- Use [putio.py](https://pypi.python.org/pypi/putio.py) package to implement the put.io client
- Support specifying put.io folder as default parent to download torrents to

- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read [contribution guide](https://github.com/SickRage/SickRage/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked

- [x] Give a description on what the PR is for.
- [ ] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

**Comment by ukanga** [comment]
Did PR in the wrong branch, apologies

**Comment by sharkykh** [comment]
you can just change it by editing the pr

File: setup.py [hunk 1]
```diff
@@ -31,7 +31,8 @@
         'pytz',
         'requests',
         'mako',
-        'configobj'
+        'configobj',
+        'putio.py'
```
Comment (sharkykh) [inline setup.py]: putio.py should be installed using pip into `lib` folder instead
In the SickRage directory, run: `pip install putio.py -t lib`, also delete *.egg file afterwards.

File: sickbeard/clients/putio_client.py [hunk 2]
```diff
@@ -15,8 +15,8 @@
 # You should have received a copy of the GNU General Public License
 # along with SickRage.  If not, see <http://www.gnu.org/licenses/>.
 
-import re
-from requests.compat import urlencode, urljoin, urlparse
+from putiopy import Client as PutioClient
+from putiopy import ClientError
```
Comment (sharkykh) [inline sickbeard/clients/putio_client.py]: Could be
```python
from putiopy import Client as PutioClient, ClientError
```

**Review by sharkykh — DISMISSED** [review]
Thanks for your contribution!

I can't test this, so I won't comment on the implementation, just a couple of general notes.

File: setup.py (hunk 1 above)
Comment (ukanga) [reply setup.py]: Noted, will make changes shortly.

File: sickbeard/clients/putio_client.py (hunk 2 above)
Comment (ukanga) [reply sickbeard/clients/putio_client.py]: I have updated the code with the changes as requested.

File: setup.py (hunk 1 above)
Comment (miigotu) [reply setup.py]: I do want to start making a list of these libs to be able to add them to setup.py in the future and make it possible to release on pypi

**Review by miigotu — APPROVED** [review]
Looks fairly straight forward, but I cannot test (no-one can without a paid account, and mine is expired)

**Comment by miigotu** [comment]
@sharkykh we should disable that binsearch test for now, maybe we can find a better public indexer to test with, or add some api keys to the CI's to run tests on private indexers

**Comment by miigotu** [comment]
Fixes #3309 
Fixes #2715

**Pull request merged**